src/lib/containers/entity/annotations/CustomSelectWidget.tsx

```diff
@@ -0,0 +1,104 @@
+import { Widget, WidgetProps } from '@sage-bionetworks/rjsf-core'
+import React from 'react'
+import Creatable from 'react-select/creatable'
+import {
+  components,
+  ControlProps,
+  OptionTypeBase,
+  GroupTypeBase,
+} from 'react-select'
+
+// Some types for rjsf and react-select don't seem to be accurate or useful, so we cast to this object when we're dealing with enumeration options
+type EnumOption = { value: string; label: string }
+
+/**
+ * We want to apply the 'form-control' bootstrap class to react-select's Control component, and the easiest way to do that is to make a custom version
+ */
+const Control = ({
+  children,
+  ...rest
+}: ControlProps<OptionTypeBase, boolean, GroupTypeBase<OptionTypeBase>>) => {
+  return (
+    <components.Control {...rest} className="form-control">
+      {children}
+    </components.Control>
+  )
+}
+
+/**
+ * react-select's value prop should be an EnumOption (see custom type defined above)
+ *
+ * This function finds the enum option that has the corresponding value and returns it.
+ *
+ * If value is nullish, this fn returns undefined. If there is no corresponding option, then a new object is returned
+ * where the label and value are set to the provided value.
+ */
+function findValueOption(
+  value: string | null | undefined,
+  options: EnumOption[],
+): EnumOption | undefined {
+  if (value == null) {
+    return undefined
+  }
+
+  const correspondingOption = options.filter(option => option.value === value)
+  if (correspondingOption.length > 0) {
+    return correspondingOption[0]
+  } else {
+    return { value: value, label: value }
+  }
+}
+
+/**
+ * Select widget compatible with react-jsonschema-form enumerations
+ */
+export const CustomSelectWidget: Widget = (props: WidgetProps) => {
+  const {
+    id,
+    options,
+    value,
+    required,
+    disabled,
+    readonly,
+    multiple,
+    autofocus,
+    onChange,
+    onBlur,
+    onFocus,
+    placeholder,
+  } = props
+  const { enumOptions } = options as {
+    enumOptions: EnumOption[]
+  }
+  return (
+    <Creatable
```

Comment:
It's like you've implemented the adapter pattern to use react-select in rjsf-core.  Nice work!